cli/client.go

```diff
@@ -418,6 +422,11 @@ func isProdBaseURL(baseURL *url.URL) bool {
 }
 
 func newAppClient(c *cli.Context) (*appClient, error) {
+	// Check for injectedAppClientKey for testing.
+	if ac := c.Context.Value(injectedAppClientKey{}); ac != nil {
+		return ac.(*appClient), nil
+	}
+
```

Comment:
I don't particularly like injecting app into the context like this but I'm struggling to think of a better way as well

Reply:
For sure. The main issue is that I think we want to test the `Action`s, but want to avoid the auth logic with `app.viam.com`. `Action`s only take one param by definition: a `cli.Context`. For testing purposes, we need to communicate through that single parameter that we (a) don't want to do auth and (b) want to use an injected app client with mock methods on its `AppServiceClient` and (probably soon) `DataServiceClient`. The only other solution that comes to mind is one with global state: relying on a globally declared `appClient` instead of one embedded into the context. Global state is definitely frowned upon in testing (tests can unintentionally affect each other), but storing anything on a context is also "frowned upon" generally 🤷🏻 .

Reply:
Agreed this seems strange, not sure I have a better solution.

Reply:
Yep that works! Great idea. I don't _love_ that we now have stuff like `LoginAction`, `loginAction` and `login`, but 🤷🏻 seems better than context injecting.